parsers/JP.py

```diff
@@ -0,0 +1,132 @@
+        }
+    df = fetch_consumption(zone_key, target_datetime)
+    df['imports'] = 0
+    for zone in exch_map[zone_key]:
+        df2 = occtonet.fetch_exchange(zone_key, zone, target_datetime)
+        df2 = pd.DataFrame(df2)
+        exchname = df2.loc[0, 'sortedZoneKeys']
+        df2 = df2[['datetime', 'netFlow']]
+        df2.columns = ['datetime', exchname]
+        df = pd.merge(df, df2, how='inner', on='datetime')
+        if exchname.split('->')[-1] == zone_key:
+            df['imports'] = df['imports']+df[exchname]
+        else:
+            df['imports'] = df['imports']-df[exchname]
+    df['prod'] = df['cons']-df['imports']
+    df = df[['datetime', 'prod']]
+    # add a row to production for each entry in the dictionary:
+    sources = {
+        'JP-HKD':'denkiyoho.hepco.co.jp',
+        'JP-TH':'setsuden.tohoku-epco.co.jp',
+        'JP-TK':'www.tepco.co.jp',
+        'JP-CB':'denki-yoho.chuden.jp',
+        'JP-HR':'www.rikuden.co.jp/denki-yoho',
+        'JP-KN':'www.kepco.co.jp',
+        'JP-SK':'www.energia.co.jp',
+        'JP-CG':'www.yonden.co.jp'
+        }
+    datalist = []
+    for i in range(df.shape[0]):
+        data = {
+            'zoneKey': zone_key,
+            'datetime': df.loc[i, 'datetime'],
+            'production': {
+                'biomass': None,
+                'coal': None,
+                'gas': None,
+                'hydro': None,
+                'nuclear': None,
+                'oil': None,
+                'solar': None,
+                'wind': None,
+                'geothermal': None,
+                'unknown': df.loc[i, 'prod']
+            },
+            'storage': {},
+            'source': ['occtonet.or.jp', sources[zone_key]]
+            }
+        datalist.append(data)
+    return datalist
+
+
+def fetch_consumption(zone_key='JP-TK', target_datetime=None,
+                      logger=logging.getLogger(__name__)):
+    """
+    Returns the consumption for an area as a pandas DataFrame
+    """
+    datestamp = arrow.get(target_datetime).to('Asia/Tokyo').strftime('%Y%m%d')
+    consumption_url = {
+        'JP-HKD': "".join(['http://denkiyoho.hepco.co.jp/area/data/juyo_01_',
+                           datestamp, '.csv']),
+        'JP-TH': "".join(['http://setsuden.tohoku-epco.co.jp/common/demand/juyo_02_',
+                          datestamp, '.csv']),
+        'JP-TK': 'http://www.tepco.co.jp/forecast/html/images/juyo-j.csv',
+        'JP-HR': "".join(['http://www.rikuden.co.jp/denki-yoho/csv/juyo_05_', datestamp, '.csv']),
+        'JP-CB': 'http://denki-yoho.chuden.jp/denki_yoho_content_data/juyo_cepco003.csv',
+        'JP-KN': 'http://www.kepco.co.jp/yamasou/juyo1_kansai.csv',
+        'JP-CG': "".join(['http://www.energia.co.jp/jukyuu/sys/juyo_07_', datestamp, '.csv']),
+        'JP-SK': 'http://www.yonden.co.jp/denkiyoho/juyo_shikoku.csv'
+        }
+    if zone_key == 'JP-KN':
+        startrow = 44
+    else:
+        startrow = 42
```

Comment:
I would add a brief comment explaining why this number of rows is being skipped (even if it's just "this is the weird format of the data sources")